WCF/Shared/Implementation/WcfOperationContext.cs

```diff
@@ -107,14 +112,37 @@ void IExtension<OperationContext>.Detach(OperationContext owner)
 
         private String DiscoverOperationName(OperationContext operationContext)
         {
+            var runtime = operationContext.EndpointDispatcher.DispatchRuntime;
             String action = operationContext.IncomingMessageHeaders.Action;
-            foreach ( var op in operationContext.EndpointDispatcher.DispatchRuntime.Operations )
+            if ( !String.IsNullOrEmpty(action) )
+            {
+                foreach ( var op in runtime.Operations )
+                {
+                    if ( op.Action == action )
+                        return op.Name;
```

Comment:
nitpick: It's good practice to always use braces { } even when the compiler doesn't require it for one liners.